src/models/principal.js

```diff
@@ -100,6 +109,18 @@ export class Principal extends DavObject {
 
 					return this.url.split('/').splice(-2, 2)[this.url.endsWith('/') ? 0 : 1];
 				}
+			},
+			roomAddress: {
```

Comment:
Is this needed here? I feel it would be more appropriate to let the clients manage room text representation.

Reply:
Good point. I just wanted to prevent boilerplate code and those are just a few lines so I guess it's fine.